docs/router/traffic-shaping/circuit-breaker.mdx

```diff
@@ -0,0 +1,174 @@
+
+### State Transition Logic
+
+The circuit breaker's state transitions follow a carefully designed logic that balances protection with availability.
+
+**Transition from Closed to Open**: The circuit breaker will only transition from closed to open when both of two critical conditions are met simultaneously. First, the minimum number of requests specified by `request_threshold` must have been received within the rolling window. Second, the error rate must exceed the percentage defined by `error_threshold_percentage`. This dual-condition approach is crucial because it prevents the circuit from opening due to a few isolated failures when there isn't enough data to make a reliable decision. For example, even if you have a 100% error rate, the circuit won't open until the request threshold is met, preventing premature circuit opening during low-traffic periods.
+
+**Transition from Open to Half-Open**: This transition happens automatically after the `sleep_window` duration expires. The circuit breaker doesn't require any external trigger—it simply moves to the half-open state to begin testing whether the downstream service has recovered.
+
+**Transition from Half-Open to Closed or Open**: From the half-open state, the circuit can transition in two directions. If the required number of successful requests (as defined by `required_successful`) are achieved during the testing phase, the circuit transitions back to closed, allowing normal traffic flow to resume. However, if any of the test requests fail, the circuit immediately returns to the open state and waits for another sleep window before attempting to test recovery again.
+
+## Example YAML Configuration
+
+```yaml
+traffic_shaping:
+  all:
+    circuit_breaker:
+      enabled: true
+      request_threshold: 20           # Need 20+ requests before evaluating
+      error_threshold_percentage: 50  # Open circuit at 50% error rate
+      sleep_window: 30s               # Block requests for 30 seconds
+      half_open_attempts: 5           # Allow 5 test requests
+      required_successful: 3          # Need 3 successes to close circuit
+      rolling_duration: 60s           # 60-second evaluation window
+      num_buckets: 10                 # 10 buckets = 6 seconds per bucket
+      execution_timeout: 60s          # Max time before marking as error
+  subgraphs:
+    employees:
+      circuit_breaker:
+        enabled: false  # Disable for this specific subgraph
+    products:
+      circuit_breaker:
+        enabled: true
+        request_threshold: 30         # Override global setting
+```
+
+## Configuration Options
```

Comment:
Do we really need to repeat what already is documented in the router config? We should use this page to guide the user through different practical configuration. We also extend the semantics but we should try to explain circuit breaker in detail.

Reply:
I think a few other places has config details in the docs, and was following the same pattern; but this makes sense, will refer them to the router config.

Reply:
Updated to refer to router config page.